src/Context/Context.tsx

```diff
@@ -0,0 +1,262 @@
+  // #region variables
+
+  const { pathname } = useLocation();
+  const isStartWithPeop = pathname.startsWith('/people');
+
+  const peopleByQuery = useMemo(() => {
+    return people.filter(person => {
+      const { name, motherName, fatherName } = person;
+      const LCName = name.toLowerCase();
+      const LCMotherName = motherName?.toLowerCase();
+      const LCFatherName = fatherName?.toLowerCase();
+      const LCQuery = query.toLowerCase();
+      const isQueryIncluded =
+        LCName.includes(LCQuery) ||
+        LCMotherName?.includes(LCQuery) ||
+        LCFatherName?.includes(LCQuery);
+
+      if (isQueryIncluded) {
+        return person;
+      }
+
+      return;
+    });
+  }, [people, query]);
+
+  const peopleByCenturies = useMemo(() => {
+    return peopleByQuery.filter(person => {
+      const { born, died } = person;
+      const bornCentury = String(born + 100).slice(0, 2);
+      const diedCentury = String(died + 100).slice(0, 2);
+
+      if (
+        activeCenturies.length === 0 ||
+        activeCenturies.includes(bornCentury) ||
+        activeCenturies.includes(diedCentury)
+      ) {
+        return person;
+      }
+
+      return;
+    });
+  }, [peopleByQuery, activeCenturies]);
+
+  const filteredPeople = useMemo(() => {
+    const resultingList = peopleByCenturies.filter(person => {
+      switch (filter) {
+        case Filter.female:
+          return person.sex === 'f';
+        case Filter.male:
+          return person.sex === 'm';
+        default:
+          return person;
+      }
+    });
+
+    if (sortParam) {
+      const sortResList = resultingList.toSorted((personA, personB) => {
+        switch (sortParam) {
+          case SortValues.name:
+            return personA.name.localeCompare(personB.name);
+          case SortValues.sex:
+            return personA.sex.localeCompare(personB.sex);
+          case SortValues.born:
+            return personA.born - personB.born;
+          case SortValues.died:
+            return personA.died - personB.died;
+        }
+      });
+
+      if (orderParam === 'desc') {
+        return sortResList.toReversed();
+      }
+
+      return sortResList;
+    }
+
+    return resultingList;
+  }, [peopleByCenturies, sortParam, filter, orderParam]);
+
+  // #endregion
+  // #region handlers
+
+  const queryHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setQuery(value);
+
+    if (value.length === 0) {
+      searchParams.delete('query');
+    } else {
+      searchParams.set('query', value);
+    }
+
+    setSearchParams(searchParams);
+  };
+
+  const centuriesHandler = (value: string) => {
+    if (value.length === 0) {
+      setActiveCenturies([]);
+      searchParams.delete('centuries');
+    } else if (activeCenturies.includes(value)) {
+      setActiveCenturies(centuries =>
+        centuries.filter(century => century !== value),
+      );
+      searchParams.delete('centuries', `${value}`);
```

Comment:
The `delete` method of `URLSearchParams` does not accept a second argument. Please remove the second argument from `searchParams.delete('centuries', `${value}`);` to avoid runtime errors.